core/templates/pages/signup-page/signup-page.component.ts

```diff
@@ -176,15 +180,11 @@ export class SignupPageComponent {
 
     let requestParams = {
       agreed_to_terms: agreedToTerms,
-      can_receive_email_updates: null,
+      can_receive_email_updates: false,
       default_dashboard: defaultDashboard,
-      username: null
+      username: username
     };
 
-    if (!this.hasUsername) {
-      requestParams.username = username;
-    }
```

Comment:
Where is this function used? Are we always expecting username to be present?

Reply:
Removed, this check as we already check this condition, so directly assign requestParams.username to username, above.